lib/cocoapods/target/build_settings.rb

```diff
@@ -1246,6 +1271,11 @@ def user_target_xcconfig_values_by_consumer_by_key
           merged_xcconfigs(user_target_xcconfig_values_by_consumer_by_key, :user_target_xcconfig)
         end
 
+        # @see BuildSettings#should_apply_xctunwrap_fix?
+        def should_apply_xctunwrap_fix?
+          target.library? && target.user_targets.any? { |ut| Version.new(ut.deployment_target) <= Version.new('12.1') }
```

Comment:
I am open to different heuristics on _when_ to apply this fix for aggregate targets.

The same question also applies for pod targets which I am implementing as we speak and will update the PR when its ready.

Is the answer "we should apply this always" ?

Reply:
Using the deployment target seems reasonable to me, as it will at least phase out the fix over time. The only alternative that I see would be checking the version of the currently selected version of Xcode and only apply it if Xcode 11 is selected, but I'm not sure there is any precedent for doing that.

Reply:
I think we could tighten this up only be applied to libraries that explicitly define a dependency on XCTest in their podspec via an explicit `s.framework = 'XCTest'`? I think that effectively matches what is called out as the problem in Apple's documentation. That would prevent adding a bunch of additional search paths that aren't used to every library that gets built.

Reply:
Yes. I think the rules are:

1) Is library/framework
2) Deployment target < 12.1
3) XCTest is a framework linked.